shared/src/telemetry/bgworker.rs

```diff
@@ -0,0 +1,78 @@
+use pgrx::bgworkers::{BackgroundWorker, BackgroundWorkerBuilder, SignalWakeFlags};
+use pgrx::*;
+use std::ffi::CStr;
+use std::process;
+use std::thread;
+use std::time::{Duration, Instant};
+
+use crate::telemetry::posthog::read_and_send_telemetry_data;
+
+#[pg_guard]
+pub fn setup_telemetry_background_worker(extension_name: String) {
+    // A background worker to read and send telemetry data to PostHog.
+    BackgroundWorkerBuilder::new(&format!("{}_telemetry_worker", &extension_name))
+        // Must be the name of a function in this file.
+        .set_function("telemetry_worker")
+        // Must be the name of the extension it will be loaded from.
```

Comment:
Do we really need comments for every line of code? This is all documented by pgrx

Reply:
I just copied was Neil did 🤷‍♂️ I'm cool with it either way, this is how it's done in `pg_bm25`